libraries/AP_NavEKF2/AP_NavEKF2_Control.cpp

```diff
@@ -160,13 +160,17 @@ void NavEKF2_core::setAidingMode()
     // Determine when to commence aiding for inertial navigation
     // Save the previous status so we can detect when it has changed
     prevIsAiding = isAiding;
-    // Don't allow filter to start position or velocity aiding until the tilt and yaw alignment is complete
-    bool filterIsStable = tiltAlignComplete && yawAlignComplete;
-    // If GPS usage has been prohiited then we use flow aiding provided optical flow data is present
-    bool useFlowAiding = (frontend->_fusionModeGPS == 3) && optFlowDataPresent();
-    // Start aiding if we have a source of aiding data and the filter attitude algnment is complete
-    // Latch to on
-    isAiding = ((readyToUseGPS() || useFlowAiding) && filterIsStable) || isAiding;
+    // perform aiding checks if not aiding
+     if (!isAiding) {
+        // Don't allow filter to start position or velocity aiding until the tilt and yaw alignment is complete
+        // and IMU gyro bias estimates have stabilised
+        bool filterIsStable = tiltAlignComplete && yawAlignComplete && checkGyroCalStatus();
+        // If GPS usage has been prohiited then we use flow aiding provided optical flow data is present
+        bool useFlowAiding = (frontend->_fusionModeGPS == 3) && optFlowDataPresent();
+        // Start aiding if we have a source of aiding data and the filter attitude algnment is complete
+        // Latch to on
+        isAiding = (readyToUseGPS() || useFlowAiding) && filterIsStable;
+    }
```

Comment:
@OXINARF  I've done another pass through the logic and cleaned it up. The fusion mode transitions now all occur in the one control function.  It transitions correctly in sitl and on the bench when GPS is disconnected. I will do some flight testing on 95e9f07 this weekend.